File: customview/src/main/java/dinson/customview/adapter/_009ContentAdapter.java
package dinson.customview.adapter;

import android.content.Context;
import android.view.View;

import java.util.List;

import dinson.customview.R;
import dinson.customview.model._009PanoramaImageModel;
import dinson.customview.utils.GlideUtils;
import dinson.customview.utils.LogUtils;
import dinson.customview.weight.TasksCompletedView;
import dinson.customview.weight.recycleview.CommonAdapter;
import dinson.customview.weight.recycleview.CommonViewHolder;

/**
 * @author Dinson - 2017/7/21
 */
public class _009ContentAdapter extends CommonAdapter<_009PanoramaImageModel> {


    public _009ContentAdapter(Context context, List<_009PanoramaImageModel> dataList) {
        super(context, dataList);
    }

    @Override
    public int getLayoutId(int viewType) {
        return R.layout.item_009_panorana_content;
    }


    @Override
    public void convert(CommonViewHolder holder, _009PanoramaImageModel bean, int position) {
        holder.setTvText(R.id.tvTitle, bean.title);
        holder.setTvText(R.id.tvDesc, bean.desc);
        GlideUtils.setImage(mContext, bean.smallImg, holder.getView(R.id.ivImg));

        TasksCompletedView progress = holder.getView(R.id.progress);
        int current = (int) (bean.getProgress() * 100);

        LogUtils.e("current: " + current);

        progress.setVisibility((current == 0 || current == 100) ? View.INVISIBLE : View.VISIBLE);
        holder.getView(R.id.complete).setVisibility(current == 100 ? View.VISIBLE : View.INVISIBLE);
        progress.setProgress(current);
    }
}

File: blingbling/src/main/java/dinson/customview/adapter/_009ContentAdapter.kt
package dinson.customview.adapter

import android.view.View
import dinson.customview.R
import dinson.customview.kotlin.logi
import dinson.customview.model._009PanoramaImageModel
import dinson.customview.utils.GlideUtils
import com.dinson.blingbase.widget.recycleview.CommonAdapter
import com.dinson.blingbase.widget.recycleview.CommonViewHolder
import kotlinx.android.synthetic.main.item_009_panorana_content.view.*

/**
 * googleVR列表数据适配器
 */
class _009ContentAdapter(dataList: MutableList<_009PanoramaImageModel>)
    : CommonAdapter<_009PanoramaImageModel>(dataList) {

    override fun getLayoutId(viewType: Int): Int = R.layout.item_009_panorana_content


    override fun convert(holder: CommonViewHolder, bean: _009PanoramaImageModel, position: Int) {
        holder.itemView.tvTitle.text = bean.title
        holder.itemView.tvDesc.text = bean.desc
        GlideUtils.setImage(holder.itemView.context, bean.smallImg, holder.itemView.ivImg)

        val current = (bean.progress * 100).toInt()

        logi { "current: $current" }

        holder.itemView.progress.visibility = if (current == 0 || current == 100) View.INVISIBLE else View.VISIBLE
        holder.itemView.complete.visibility = if (current == 100) View.VISIBLE else View.INVISIBLE
        holder.itemView.progress.setProgress(current)
    }
}
